### PR TITLE
Add Ollama as a provider in proxy ui

### LINKED ISSUE #4018: [Bug]: Authentication Error, No module named 'nacl'.
### What happened?

Steps to reproduce.

1. Start litellm proxy using vscode dev container.
2. navigate to models in ui and try adding a model via the ui.

Screenshot_5-6-2024_152140_localhost

### Relevant log output

```shell
###### Error shown in UI ###############################################################
Failed to create key: {"error":{"message":"Authentication Error, No module named 'nacl'","type":"auth_error","param":"None","code":400}}

###### Server Output ###################################################################
Traceback (most recent call last):
  File "/workspaces/litellm/litellm/proxy/proxy_server.py", line 11308, in add_new_model
    encrypted_value = encrypt_value(value=v, master_key=master_key)  # type: ignore
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/litellm/litellm/proxy/utils.py", line 2674, in encrypt_value
    import nacl.secret
ModuleNotFoundError: No module named 'nacl'
```

### Twitter / LinkedIn details

_No response_

### LINKED ISSUE #4019: [Feature]: Add Ollama as a provider in the proxy UI
### The Feature

I would like Ollama to be available as a provider via the proxy UI.
In the UI Models > Add Models > Providers

### Motivation, pitch

Currently you can only add ollama models via the yaml config file and you need to restart the server everytime you do this.

### Twitter / LinkedIn details

_No response_

### PR DESCRIPTION
**Pull request by sha-ahammed**
## Title

Added Ollama as a provider in proxy ui

## Relevant issues

1. Fixes #4018 
2. Closes #4019 

## Type

🆕 New Feature
🐛 Bug Fix

## Changes
- Added Ollama as a provider in ui/litellm_dashboard/src/components/model_dashboard.tsx
- Added get_required_params method for ollama called from get_provider_fields in litellm/utils.py
- Added pynacl to pyproject.toml and updated poetry.lock

## [REQUIRED] Testing - Attach a screenshot of any new tests passing locally

```shell
======================================================================================= short test summary info ========================================================================================
ERROR test_azure_perf.py - openai.OpenAIError: Missing credentials. Please pass one of `api_key`, `azure_ad_token`, `azure_ad_token_provider`, or the `AZURE_OPENAI_API_KEY` or `AZURE_OPENAI_AD_TOKEN` environment variables.
ERROR test_banned_keyword_list.py -  ModuleNotFoundError: No module named 'litellm.proxy.enterprise'
ERROR test_blocked_user_list.py -  ModuleNotFoundError: No module named 'litellm.proxy.enterprise'
ERROR test_config.py - ValueError: api_base is required for Azure OpenAI. Set it on your config. Model - {'model_name': 'gpt-3.5-turbo', 'litellm_params': {'model': 'azure/chatgpt-v-2'}}
ERROR test_lakera_ai_prompt_injection.py -  ModuleNotFoundError: No module named 'litellm.proxy.enterprise'
ERROR test_llm_guard.py -  ModuleNotFoundError: No module named 'litellm.proxy.enterprise'
ERROR test_logfire.py - ModuleNotFoundError: No module named 'logfire'
ERROR test_multiple_deployments.py - Failed: An exception occurred: 'NoneType' object has no attribute '_response_ms'
ERROR test_openai_moderations_hook.py - ModuleNotFoundError: No module named 'litellm.proxy.enterprise'
ERROR test_opentelemetry.py - ModuleNotFoundError: No module named 'opentelemetry'
ERROR test_register_model.py - Failed: An error occurred: AuthenticationError: OpenAIException - Traceback (most recent call last):
ERROR test_traceloop.py - ModuleNotFoundError: No module named 'opentelemetry'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 12 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

If UI changes, send a screenshot/GIF of working UI fixes

[LiteLLM Dashboard - 5 June 2024 - Watch Video](https://www.loom.com/share/478628b4edb94e019d90493b6591dab4)

### CODE REVIEW

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
can you remove the changes you made to the UI build

**Comment by sha-ahammed** [comment]
@ishaan-jaff I have updated, Can you review?

File: pyproject.toml [hunk 1]
```diff
@@ -40,6 +40,7 @@ azure-identity = {version = "^1.15.0", optional = true}
 azure-keyvault-secrets = {version = "^4.8.0", optional = true}
 google-cloud-kms = {version = "^2.21.3", optional = true}
 resend = {version = "^0.8.0", optional = true}
+pynacl = "^1.5.0"
```
Comment (ghost) [inline pyproject.toml]: can you remove this please. @sha-ahammed this is not a requirement for the sdk. it seems proxy specific.

**Review by ghost — CHANGES_REQUESTED** [review]
please remove the dependency on pynacl

File: pyproject.toml (hunk 1 above)
Comment (sha-ahammed) [reply pyproject.toml]: > can you remove this please. @sha-ahammed this is not a requirement for the sdk. it seems proxy specific.

Should I add it as a dependency for `[tool.poetry.extras]` in `pyproject.toml`

Comment (ghost) [reply pyproject.toml]: yep i think that's best

Comment (sha-ahammed) [reply pyproject.toml]: I have reverted the commit. Can we have this merged? @krrishdholakia ?

**Comment by ghost** [comment]
Yup - great PR @sha-ahammed curious - how're you using litellm today?

**Comment by ghost** [comment]
@sha-ahammed this fails linting - can you fix and bump me? 

https://github.com/BerriAI/litellm/actions/runs/9403674514/job/25901042338?pr=4020

**Comment by sha-ahammed** [comment]
> @sha-ahammed this fails linting - can you fix and bump me?
> 
> https://github.com/BerriAI/litellm/actions/runs/9403674514/job/25901042338?pr=4020

Okay give me a sec.

**Comment by sha-ahammed** [comment]
> Yup - great PR @sha-ahammed curious - how're you using litellm today?

We use AnythingLLM and had requested them to integrate LiteLLM as a Provider. We are planning on serving our models to AnythingLLM via LiteLLM.

**Comment by ghost** [comment]
Nice. DM'ed on linked to setup a support channel. 

Can we do a 10min call sometime this/next week? Would love to learn how we can improve for you 

https://calendly.com/d/4mp-gd3-k5k/litellm-1-1-onboarding-chat

**Pull request merged**